Tools/cases_generator/optimizer_generator.py

```diff
@@ -111,19 +122,88 @@ def goto_label(self, goto: Token, label: Token, storage: Storage) -> None:
         self.out.emit(goto)
         self.out.emit(label)
 
+
+class OptimizerConstantEmitter(OptimizerEmitter):
+    def __init__(self, out: CWriter, labels: dict[str, Label], uop: Uop):
+        super().__init__(out, labels)
+        # Replace all outputs to point to their stackref versions.
+        overrides = {
+            outp.name: self.emit_stackref_override for outp in uop.stack.outputs
+        }
+        self._replacers = {**self._replacers, **overrides}
+
+    def emit_stackref_override(
+        self,
+        tkn: Token,
+        tkn_iter: TokenIterator,
+        uop: CodeSection,
+        storage: Storage,
+        inst: Instruction | None,
+    ) -> bool:
+        self.out.emit(tkn)
+        self.out.emit("_stackref")
```

Comment:
I'm not sure how much we care about formatting of the generated code, but we can insert a space here to make the assignment a bit nicer (example):

```diff
-                res_stackref= PyStackRef_IsFalse(value)
+                res_stackref = PyStackRef_IsFalse(value)
```

```suggestion
        self.out.emit("_stackref ")
```